include/net/buf.h

```diff
@@ -1356,6 +1356,32 @@ u16_t net_buf_append_bytes(struct net_buf *buf, u16_t len,
 			   const u8_t *value, s32_t timeout,
 			   net_buf_allocator_cb allocate_cb, void *user_data);
 
+/**
+ * @brief Append data to a list of net_buf and fill it with the speccified byte
+ *
+ * @details appends multiple bytes to the buffer and sets them to
+ * the specified byte, just like memset. If there is not enough space in the
+ * net_buf then more net_buf will be added, unless there are no free net_buf
+ * and timeout occurs.
+ *
+ * @param buf Network buffer.
+ * @param len Total length of data to append
+ * @param value Byte to fill the buffer with
+ * @param timeout Timeout is passed to the net_buf allocator callback.
+ * @param allocate_cb When a new net_buf is required, use this callback.
+ * @param user_data A user data pointer to be supplied to the allocate_cb.
+ *        This pointer is can be anything from a mem_pool or a net_pkt, the
+ *        logic is left up to the allocate_cb function.
+ *
+ * @return Length of data actually added. This may be less than input
+ *         length if other timeout than K_FOREVER was used, and there
+ *         were no free fragments in a pool to accommodate all data.
+ */
+u16_t net_buf_append_set_bytes(struct net_buf *buf, u16_t len,
+			       const u8_t value, s32_t timeout,
```

Comment:
Looks like indentation is off here.

Reply:
Hm, looks perfectly indented here. Checkpatch also approves. Not sure how to improve that.

Reply:
Yep, my bad, no issue in this line.

Reply:
This sounds so specific that I'd rather move it to the net_pkt layer. Some comments nevertheless:

 - The verb used for adding stuff to the end of buffers is "add", not "append". I noticed that `net_buf_append_bytes()` has slipped past me. Was there a specific reason to change the verb for that API?
 - const for the `u8_t value` does nothing, so I'd leave it out.
 - It should be `size_t len` and not `u16_t len` to be consistent throughout the API (not to mention it generates less code for our 32-bit architectures). `net_buf_append_bytes()` got this wrong also - needs fixing.